source/slang/slang-check-expr.cpp

```diff
@@ -1529,6 +1538,51 @@ namespace Slang
         return CheckInvokeExprWithCheckedOperands(expr);
     }
 
+    Expr* SemanticsExprVisitor::visitTryExpr(TryExpr* expr)
+    {
+        auto prevTryClauseType = expr->tryClauseType;
+        m_enclosingTryClauseType = expr->tryClauseType;
+        expr->base = CheckTerm(expr->base);
+        m_enclosingTryClauseType = prevTryClauseType;
+        expr->type = expr->base->type;
+        if (as<ErrorType>(expr->type))
+            return expr;
+        
+        auto parentFunc = this->m_parentFunc;
+        // TODO: check if the try clause is catched.
+        // For now we assume all `try`s are not catched (because we don't have catch yet).
+        if (!parentFunc)
+        {
+            getSink()->diagnose(expr, Diagnostics::uncachedTryCallInNonThrowFunc);
+            return expr;
+        }
+        if (parentFunc->errorType->equals(m_astBuilder->getVoidType()))
+        {
+            getSink()->diagnose(expr, Diagnostics::uncachedTryCallInNonThrowFunc);
```

Comment:
I think the diagnostic names here are using "cached" and the comments are using "catched" when the word should be "caught".